packages/good-design/src/apps/goodid/wizards/WizardHeader.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useCallback } from "react";
+import { useWizard } from "react-use-wizard";
+import { TouchableOpacity } from "react-native";
+import { ArrowBackIcon, Text, View } from "native-base";
+
+import { ErrorModal } from "../../../core/web3";
+
+export const WizardHeader = ({
+  onDone,
+  onError,
+  onClose
+}: {
+  onDone: (error?: Error) => Promise<void>;
+  onError: any;
+  onClose?: any;
+}) => {
+  const { isFirstStep, previousStep, isLastStep, goToStep } = useWizard();
+
+  const handleBack = useCallback(() => {
+    if (isFirstStep) {
+      void onDone();
+      return;
+    }
+    previousStep();
+  }, [isFirstStep]);
+
+  if (onError) return <ErrorModal error={onError} onClose={onClose ?? (() => goToStep(0))} overlay="dark" />;
```

Comment:
1. maybe rename this prop to just `error` ?
it would be logically - if we pass some error, display error message instead of generic UI
`on` means some callback to be executed _if error_ occurs

2. use ternary as in the example above

Reply:
done here: https://github.com/GoodDollar/GoodWeb3-Mono/pull/136/commits/052b4a16210235518db27fa715bc843fb200237a